pkg/trait/trait.go

```diff
@@ -79,11 +79,15 @@ func newEnvironment(ctx context.Context, c client.Client, integration *v1.Integr
 
 	if kit == nil {
 		kit, err = getIntegrationKit(ctx, c, integration)
-		if err != nil {
+		if err != nil && !k8serrors.IsNotFound(err) {
 			return nil, err
 		}
 	}
 
+	//
+	// kit can still be nil if integration kit is yet
```

Comment:
The kit is assigned before the build starts.

Reply:
Not the case if the kit has been explicitly unassigned. See explanation above.